src/Google/Service.php

```diff
@@ -25,9 +25,21 @@ class Google_Service
   public $resource;
   private $client;
 
-  public function __construct(Google_Client $client)
+  /**
+   * @param Google_Client|array $clientOrConfig Optional
+   */
+  public function __construct($clientOrConfig = array())
   {
-    $this->client = $client;
+    if ($clientOrConfig instanceof Google_Client) {
+      $this->client = $clientOrConfig;
+    } elseif (is_array($clientOrConfig)) {
+      $this->client = new Google_Client($clientOrConfig ?: array());
+    } else {
+      $errorMessage = 'constructor must be array or instance of Google_Client';
+      throw class_exists('TypeError')
```

Comment:
I'm not sold on this condition, especially since the only way to catch both in a single block is using `Throwable`. Why not just use `InvalidArgumentException`?

Reply:
I am on the fence as well but figure we should throw the TypeError when we can for two reasons: 1. TypeError is the more correct thing to do and 2. it's less disruptive since this is currently throwing a TypeError.

Willing to change it back though...

Reply:
ah, that's a good point. I guess it's unlikely that somebody would be catching a bad constructor argument exception anyhow -- it's almost 100% a bug, right? not a recoverable condition, more something you fix and re-run the script.

Since in PHP 5 such a case would be a fatal error, maybe instead of throwing `InvalidArgumentException` we should trigger a fatal error if `TypeError` is undefined. I think that would maintain current behavior in both major versions of PHP.

Reply:
I can get behind that. Something like this?

```php
trigger_error($errorMessage, E_USER_ERROR);
```

Reply:
Yep